socorro/external/postgresql/base.py

```diff
@@ -41,22 +42,17 @@ def __init__(self, *args, **kwargs):
 
         """
         self.context = kwargs.get("config")
-        if hasattr(self.context, 'database'):
-            # XXX this should be replaced with connection_context instead
-            self.context.database['database_host'] = \
-                self.context.database.database_hostname
-            self.context.database['database_port'] = \
-                self.context.database.database_port
-            self.context.database['database_name'] = \
-                self.context.database.database_name
-            self.context.database['database_username'] = \
-                self.context.database.database_username
-            self.context.database['database_password'] = \
-                self.context.database.database_password
-            self.database = db.Database(self.context.database)
-        else:
-            # the old middleware
-            self.database = db.Database(self.context)
+        try:
+            self.database =  self.context.database_class(
+                self.context
+            )
+        except KeyError:
+            # some tests seem to put the database config parameters
+            # into a namespace called 'database', others do not
+            self.database =  self.context.database.database_class(
```

Comment:
Deprecation warning? 
Or perhaps just fix the tests that goes into this except block once and for all?